nibabel/spatialimages.py

```diff
@@ -866,6 +867,22 @@ def from_image(klass, img):
                      klass.header_class.from_header(img.header),
                      extra=img.extra.copy())
 
+    @classmethod
+    def is_image(klass, filename, sniff=None):
+        ftypes = dict(klass.files_types)
+        froot, ext, trailing = splitext_addext(filename, ('.gz', '.bz2'))
+        lext = ext.lower()
+
+        if lext not in ftypes.values():
+            return False, sniff
+
+        fname = froot + ftypes['header'] if 'header' in ftypes else filename
+        if not sniff:
+            with BinOpener(fname, 'rb') as fobj:
+                sniff = fobj.read(1024)
```

Comment:
What I wanted to avoid was grabbing data from the filesystem and then finding out we need more. Hard-coding was easiest, and 1kb is small enough to avoid burdening even a slow network connection, but large enough to cover all possible headers. But if there's a cleverer way of doing it, by all means.